sklearn/linear_model/_logistic.py

```diff
@@ -289,6 +295,12 @@ def _logistic_regression_path(
     # Otherwise set them to 1 for all examples
     sample_weight = _check_sample_weight(sample_weight, X, dtype=X.dtype, copy=True)
 
+    if solver == "newton-cholesky":
+        # Same as in _GeneralizedLinearRegressor.fit().
+        # This rescaling has to be done before multiplying by class_weights.
+        sw_sum = sample_weight.sum()  # needed to rescale penalty, nasty matter!
```

Comment:
My poor self will wonder what is the nasty matter in a couple of releases :). Would it be worth adding the same important note as in the GLM?

Reply:
I'll copy it here. But note that it is even a bit more complicated here: Unfortunately, in LogReg, we use `obj = sum(weights * loss)` while in GLMs we use `np.average(loss, weights=weight)`. This last form seems better for solver stopping criteria. And this last form is also used for the `newton-cholesky` solver of LogReg.

Reply:
See https://github.com/scikit-learn/scikit-learn/pull/24767/commits/3b23b6afba595bff305d5c887c0e1c92a78777bd.